plugins/ui/src/SettingsPage/PluginStoreTab/LunaStore.tsx

```diff
@@ -44,6 +44,56 @@ export const LunaStore = React.memo(({ url, onRemove }: { url: string; onRemove:
 
 	const isLocalDevStore = url === "http://127.0.0.1:3000";
 
+	const [plugins, setPlugins] = React.useState<{ plugin?: LunaPlugin; error?: string; id: string; url: string }[]>([]);
+	const [pluginsLoaded, setPluginsLoaded] = React.useState(false);
+
+	React.useEffect(() => {
+		if (!pkg) {
+			setPlugins([]);
+			setPluginsLoaded(false);
+			return;
+		}
+
+		let aborted = false;
+		setPluginsLoaded(false);
+		const promises = pkg.plugins.map(async (pluginPath) => {
+			const pluginUrl = `${url}/${isLocalDevStore ? pluginPath : pluginPath.replace(" ", ".")}`;
+			try {
+				const plugin = await LunaPlugin.fromStorage({ url: pluginUrl });
```

Comment:
We should not be loading/fetching every plugin just to search them. This needs to be changed.